### ISSUE TITLE
Polybar is rendered on top of other windows after updating from 0.9.0 to 0.9.1

### ISSUE THREAD

**Issue by lainor** [herbstluftwm/herbstluftwm#1110]
After updating to 0.9.1, polybar with `offset-y=12` and `override-direct=false` is rendered on top of  other windows

**0.9.1**
![0.9.1](https://i.imgur.com/tAg8R4S.png)

**0.9.0**
![0.9.0](https://i.imgur.com/dgQ1wAq.png)

**Comment by t-wissmann** [comment]
This is a feature actually because the pad of the monitor is adjusted automatically to leave room for your panel. There are several options to prevent this:

  * You can disable the setting `auto_detect_panels` to prevent this. (should happen in your autostart before you start polybar).
  * You enable override-redirect (it means precisely: don't let the window manager control the window)
  * Change the window type of polybar. The detection only is active for windows of type `_NET_WM_WINDOW_TYPE_DOCK` (which means "panel")

**Comment by lainor** [comment]
Neither disabling `auto_detect_panels` and enabling `override-redirect` works.
Am I overlooking something?

**Comment by t-wissmann** [comment]
> Neither disabling `auto_detect_panels` and enabling `override-redirect` works.
> Am I overlooking something?

`auto_detect_panels` must be set before the panel shows up. And enabling  `override-redirect` must work, because hlwm can't see override-redirect windows. You can run `xwininfo` and click on the polybar and look for the line
```
Override Redirect State: yes/no
```
to verify that it's enabled.

**Comment by lainor** [comment]
`Override Redirect State: yes`

and `auto_detect_panels false` was set before polybar is launched

**Comment by t-wissmann** [comment]
And if you explicitly run
```
herbstclient pad '' 0
```
and then run polybar, the gap comes back?

**Comment by lainor** [comment]
No it does not, polybar is still rendered above the other windows

**Comment by t-wissmann** [comment]
What do you mean by "above" then? I thought with "above" you mean "as in the second image". How can it be "as in the second image" when the gap does not come back?

**Comment by lainor** [comment]
Sorry my bad!  I inverted the images. Just edited the original issue.

With "above" I meant "foreground", 0.9.1 is ignoring any settings and it places polybar "on" the other windows (meaning the other windows are in background).

0.9.0 is working as excepted by reserving the panel space and creating the gap.

**Comment by t-wissmann** [comment]
> Sorry my bad! I inverted the images. Just edited the original issue.
> 
> With "above" I meant "foreground", 0.9.1 is ignoring any settings and it places polybar "on" the other windows (meaning the other windows are in background).
> 
> 0.9.0 is working as expected by reserving the panel space and creating the gap.

Ah, it makes much more sense this way around! (I was really wondering why you preferred the overlapping case over the non-overlapping case). And in this case the word "above" is perfectly appropriate.

In this case, you should disable `override-redirect` and activate `auto_detect_panels`. Can you run `xprop`, click on the polybar and paste the output here?  Maybe I get an idea why the panel is not detected correctly.

**Comment by dnnr** [comment]
As a fellow polybar user, this made me curious. I've just upgraded to 0.9.1 and can report that it works perfectly for me, with `auto_detect_panels` enabled and `override-redirect` disabled (so the defaults, basically). Finally, no more manual padding after monitor changes! :rocket:

**Comment by lainor** [comment]
> > Sorry my bad! I inverted the images. Just edited the original issue.
> > With "above" I meant "foreground", 0.9.1 is ignoring any settings and it places polybar "on" the other windows (meaning the other windows are in background).
> > 0.9.0 is working as expected by reserving the panel space and creating the gap.
> 
> Ah, it makes much more sense this way around! (I was really wondering why you preferred the overlapping case over the non-overlapping case). And in this case the word "above" is perfectly appropriate.
> 
> In this case, you should disable `override-redirect` and activate `auto_detect_panels`. Can you run `xprop`, click on the polybar and paste the output here? Maybe I get an idea why the panel is not detected correctly.

Yes, sorry again for the confusion!

**xprop**

```
_NET_WM_PID(CARDINAL) = 19179
_NET_WM_DESKTOP(CARDINAL) = 4294967295
_NET_WM_STATE(ATOM) = _NET_WM_STATE_STICKY, _NET_WM_STATE_ABOVE
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_DOCK
WM_CLASS(STRING) = "polybar", "Polybar"
WM_NAME(STRING) = "polybar-top_eDP1"
_NET_WM_STRUT_PARTIAL(CARDINAL) = 0, 0, 54, 0, 0, 0, 0, 0, 192, 1919, 0, 0
_NET_WM_STRUT(CARDINAL) = 0, 0, 54, 0
```

**xwininfo**
```

xwininfo: Window id: 0x1400001 "polybar-top_eDP1"

  Absolute upper-left X:  192
  Absolute upper-left Y:  12
  Relative upper-left X:  192
  Relative upper-left Y:  12
  Width: 1536
  Height: 42
  Depth: 32
  Visual: 0x8f
  Visual Class: TrueColor
  Border width: 0
  Class: InputOutput
  Colormap: 0x1400000 (not installed)
  Bit Gravity State: ForgetGravity
  Window Gravity State: NorthWestGravity
  Backing Store State: WhenMapped
  Save Under State: no
  Map State: IsViewable
  Override Redirect State: no
  Corners:  +192+12  -192+12  -192-1026  +192-1026
  -geometry 1536x42+192+12
```

```
$ herbstclient get auto_detect_panels
true
```

**Comment by t-wissmann** [comment]
I could reproduce it and fixed this in #1113. You can either try the git version once the PR is merged or you can patch the sources yourself if you're fine with compiling hlwm. To this end, you can insert the lines
```c++
rs.left_ = p[WmStrut::left];
rs.right_ = p[WmStrut::right];
rs.top_ = p[WmStrut::top];
rs.bottom_ = p[WmStrut::bottom];
```
in `src/panelmanager.cpp` right before
```c++
for (size_t i = 0; i < 4; i++) {
    rsTotal[i] = std::max(rsTotal[i], rs[i]);
}
```
If you don't want to recompile, you can either set the pad to 54 manually (`herbstclient pad '' 54`) or make the panel appear with `offset-y=0`, then the detection should work again (maybe you were already aware of this workaround).

**Comment by lainor** [comment]
@t-wissmann thanks a lot for this!
I think I'm going with the git version since it can be a good occasion to help with the project
